### PR TITLE
Add migration tool for carbon applications

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
Add migration tool for https://github.com/wso2/api-manager/issues/455

### CODE REVIEW

File: migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class CAppMigrator {
+
+    private static final Logger logger = Logger.getLogger(CAppMigrator.class.getName());
```
Comment (arunans23) [inline migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java:40]: static final variables should be in Block letters.

File: migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+                    copyFileToDirectory(carLocation, tmpDirectory, carFile);
+                }
+            }
+        } else {
+            exitWithError("The provided carbon application folder " + carLocation + " is empty.");
+        }
+        String[] tmpCAppsList = new File(tmpDirectory).list();
+        if (null != tmpCAppsList) {
+            for (String file : tmpCAppsList) {
+                String carFileDir = tmpDirectory + File.separator + file;
```
Comment (arunans23) [inline migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java:72]:
```suggestion
                String carFile = tmpDirectory + File.separator + file;
```

File: migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+                String cAppFolder = unzipFile(carFileDir);
+                ArrayList<String> styleSheetList = getXsltStyleSheets(cAppFolder);
+                if (styleSheetList.size() > 0) {
+                    logger.info("xsltStyleSheet file is available in " +  file + ". Hence performing migration");
```
Comment (arunans23) [inline migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java:76]: Shall we include the number of xsltStyleSheet available in the log?

File: migration/capp-migration-service/pom.xml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (arunans23) [inline migration/capp-migration-service/pom.xml:3]: According to the new policy, it should be WSO2 LLC. Please check other locations as well.

File: migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java (hunk 1 above)
Comment (dulanjalidilmi) [reply migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java]: Fixed.

File: migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java (hunk 2 above)
Comment (dulanjalidilmi) [reply migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java]: Fixed.

File: migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+import java.util.Objects;
+import java.util.logging.Logger;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class CAppMigrator {
```
[the remark this answers is not in the document]
Comment (dulanjalidilmi) [reply migration/capp-migration-service/src/main/java/org/wso2/mi/car/migration/CAppMigrator.java]: Changed the class name to `DataMapperConfigMigrator`

File: migration/capp-migration-service/pom.xml (hunk 4 above)
Comment (dulanjalidilmi) [reply migration/capp-migration-service/pom.xml]: Fixed.

**Review by arunans23 — APPROVED** [review]

**Pull request merged**